crypto/threads_pthread.c

```diff
@@ -538,10 +558,13 @@ static struct rcu_qp *update_qp(CRYPTO_RCU_LOCK *lock)
      * of this update are published to the read side prior to updating the
      * reader idx below
      */
-    ATOMIC_AND_FETCH(&lock->qp_group[current_idx].users, ID_MASK,
-                     __ATOMIC_RELEASE);
-    ATOMIC_OR_FETCH(&lock->qp_group[current_idx].users, new_id,
-                    __ATOMIC_RELEASE);
+try_again:
+    ret = ATOMIC_LOAD_N(uint64_t, &lock->qp_group[current_idx].users, __ATOMIC_ACQUIRE);
+    update = ret & ID_MASK;
+    update |= new_id;
+    if (!ATOMIC_COMPARE_EXCHANGE_N(uint64_t, &lock->qp_group[current_idx].users, &ret, update,
+                                   __ATOMIC_ACQ_REL, __ATOMIC_RELAXED))
+        goto try_again;
```

Comment:
can you add a counter to goto branch so we will be able to see how many retries is needed. I mean this would be just for testing purposes. this will allow us to see you fix in action. the counter can be let go after successful test. printing the counter to stderr value once test application finishes.

Reply:
I can, but I have a debugging infrastucture coded up here locally that I was trying to avoid adding to not make this PR overly complex.  I was planning on itroducing that in a separate PR if thats ok with you

Reply:
I was just asking for interim change to see your fix in action. 

I understand now that that the type of test I was asking for did work. I'm fine with those changes.